privacyidea/api/lib/prepolicy.py

```diff
@@ -369,6 +369,35 @@ def check_otp_pin(request=None, action=None):
     return True
 
 
+def check_application_tokentype(request=None, action=None):
+    """
+    This pre policy checks if the request is allowed to specify the tokentype.
+    If the policy is not set, a possibly set parameter "type" is removed
+    from the request.
+
+    Check ACTION.APPLICATION_TOKENTYPE
+
+    This decorator should wrap
+        /validate/check, /validate/samlcheck and /validate/triggerchallenge.
+
+    :param request: The request that is intercepted during the API call
+    :type request: Request Object
+    :param action: An optional Action
+    :type action: basestring
+    :returns: Always true. Modified the parameter request
+    """
+    application_allowed = Match.generic(g, scope=SCOPE.AUTHZ,
+                                        action=ACTION.APPLICATION_TOKENTYPE,
+                                        user_object=request.User,
+                                        active=True).any()
+
+    # if the application is not allowed, we remove the tokentype
```

Comment:
this is a general policy. Better write "if applications are not allowed, we remove the tokentype"

Reply:
done